### PR TITLE
Refactor kubelet stats serving

### PR DESCRIPTION
**Pull request by timstclair**
Changes include:
- Moving stats serving & routes to pkg/kubelet/server/stats/handler.go
- Managing the routes with restful.WebService, rather than manual
  parsing
- Misc cleanup

These changes will make adding the new routes for /stats/summary more
manageable.

/cc @vishh

### CODE REVIEW

File: pkg/kubelet/server/stats/handler.go [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+// handleError serializes an error object into an HTTP response.
+func handleError(response *restful.Response, err error) {
+	switch err {
+	case kubecontainer.ErrContainerNotFound:
+		response.WriteError(http.StatusNotFound, err)
+	default:
+		msg := fmt.Sprintf("Internal Error: %v", err)
+		glog.Infof("HTTP InternalServerError: %s", msg)
+		response.WriteErrorString(http.StatusNotFound, msg)
```
Comment (yujuhong) [inline pkg/kubelet/server/stats/handler.go]: Shouldn't this be `http.StatusInternalServerError`?

Comment (timstclair) [inline pkg/kubelet/server/stats/handler.go]: Good catch, I don't know how that got changed.

**Comment by timstclair** [comment]
Fixed test & error code.

**Comment by yujuhong** [comment]
LGTM. This is miles better than the string parsing :)

**Comment by timstclair** [comment]
@k8s-bot unit test this
@k8s-bot test this

**Comment by timstclair** [comment]
@k8s-bot unit test this

**Comment by dchen1107** [comment]
LGTM. Nice refactory!

File: pkg/kubelet/server/stats/handler.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+		Produces(restful.MIME_JSON)
+
+	endpoints := []struct {
+		path    string
+		handler restful.RouteFunction
+	}{
+		{"", h.handleStats},
+		{"/container", h.handleSystemContainer},
+		{"/{podName}/{containerName}", h.handlePodContainer},
+		{"/{namespace}/{podName}/{uid}/{containerName}", h.handlePodContainer},
```
Comment (vishh) [inline pkg/kubelet/server/stats/handler.go]: :100:

**Comment by vishh** [comment]
LGTM @timstclair!

**Comment by timstclair** [comment]
@k8s-bot test this

**Comment by gmarek** [comment]
unit failure: #19466
@k8s-bot unit test this

**Pull request merged**